### ISSUE TITLE
Edit or Paste Message with hidden Link crashes Unigram

### ISSUE THREAD

**Issue by hihain** [UnigramDev/Unigram#1931]
Some bots use "invisible Unicode" to create an invisible link to include an image without having the user to see the link to the image or send a real photo with the message.

Pasting works, but on insert Unigram crashes on this line (same for edit, same code - twice in the same file - FormattedTextBox; SetText & InsertText):
`range.Link = $"\"{textUrl.Url}\"";`

Exception:

> Catastrophic failure (Exception from HRESULT: 0x8000FFFF (E_UNEXPECTED))
Source: Windows
   at Windows.UI.Text.RichEditTextRange.put_Link(String value)
   at Unigram.Controls.FormattedTextBox.InsertText(String text, IList`1 entities)
   at Unigram.Controls.Chats.ChatTextBox.<OnPaste>d__13.MoveNext()

Other Telegram clients seem to ignore invisible Unicode and the link for that either already on copy, or on paste. On Edit they keep the invisible link.

**Comment by hihain** [comment]
I referenced a commit. The copy & paste part is easy to hack (fix) by adding a try..catch. Because of emojis I found a try..catch easier than an if like
` && range.Text.Count(c => char.IsLetterOrDigit(c) || char.IsSymbol(c) || char.IsPunctuation(c)) > 0` + code to find emojis and perhaps other string fragments / chars that do not crash on link creation that I did not think of yet.

However, I do not like that hack much and for editing is is even worse. I do not know what the problem of UWP/Windows is and other Telegram clients are able to edit messages containing hacky invisible links (e.g. empty space, tab, or 8203/8204...). Perhaps it would need a complete different approach? Not sure if it is worth it...
